extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/TracerUtil.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.opentelemetry.runtime.tracing;
+
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+
+public final class TracerUtil {
+    private TracerUtil() {
+    }
+
+    public static Resource mapResources(List<String> resources) {
+        Map<String, ResourceProvider> spiResources = StreamSupport
+                .stream(ServiceLoader.load(ResourceProvider.class).spliterator(), false)
+                .collect(Collectors.toMap(
+                        k -> k.getClass().getSimpleName().toLowerCase().replace("resourceprovider", ""),
+                        v -> v));
+
+        return resources.stream()
```

Comment:
Using the SPI to get the resources, the down side is that right now we are passing `null` to the `createResource` as the `configProperties` parameter, all the resources doesn't use this parameter they just do `return resourceName.get()`.... and if that changes and they start to use that param something might break... @kenfinnigan do you think that resorting to beans instead will be a better idea?

Reply:
I am leaning towards the multi bean implementation + priority... what do you think?

Reply:
I think it makes sense to go with key/value pairs as config, for reading from environment variables especially, and then basing the rest off Beans of type `Resource`.

What are you thinking in terms of priority?

Reply:
Did dome tests and either the key/value option and the Resource beans needs to happen during run time, and we are creating the sdk provider using static init, is that OK to change it to `Runtime`?

Reply:
I'd prefer to keep as much as possible as `STATIC`.

One option is a similar approach to what we have for `BatchSpanProcessor` being installed into the SDK at static init time, but the actual instance is only installed into it at runtime.

So we can have a `DelayedResource` instance which is installed into the SDK, which incorporates all the `Resource` bean instances, and then at runtime, we collect the runtime config key/values and set them onto `DelayedResource` and merge them with the existing resources

Reply:
The Resource has a private constructor, so it prevents subclassing, though I was able to circumvent this by creating a delayed `Attributes` so the resource is created with the delayed attributes and the dynamic attributes get set during the runtime init